src/dataset.rs

```diff
@@ -40,11 +42,22 @@ pub struct FSRSBatch<B: Backend> {
 
 impl<B: Backend> Batcher<FSRSItem, FSRSBatch<B>> for FSRSBatcher<B> {
     fn batch(&self, items: Vec<FSRSItem>) -> FSRSBatch<B> {
+        let pad_size = items
+            .iter()
+            .map(|x| x.reviews.len())
+            .max()
+            .expect("FSRSItem is empty");
+
         let t_historys = items
             .iter()
             .map(|item| {
                 Data::new(
-                    item.reviews.iter().map(|r| r.delta_t).collect(),
+                    item.reviews
+                        .iter()
+                        .map(|r| r.delta_t)
+                        .chain(repeat(0))
+                        .take(pad_size)
+                        .collect(),
                     Shape {
                         dims: [item.reviews.len()],
```

Comment:
should that be pad_size?

Reply:
I think `pad_size` should be `max_seq_len - real_seq_len`

Reply:
.take() will take a maximum of pad_size items from original + zeros, so the iterator code part looks ok